### PR TITLE
solve button send me the link is not up to date

### PR DESCRIPTION
**Pull request by Deeppjp116**
## Description
This pull request adds email validation and disables the button unless the email contains characters, numbers, and ends with "@gmail.com".

[Screencast from 18-06-23 02:23:30 PM IST.webm](https://github.com/scorelab/Codelabz/assets/128173785/eab8ddef-9338-445a-a384-cbc0bfcf3aea)
## Changes Made
- Updated the regular expression to match email addresses containing "@gmail.com".
- Updated the button disabling logic based on the email format.
- Added email input field and form submission functionality.

## Testing
- Tested the email validation with various scenarios, including valid and invalid email formats.
- Tested the button disabling behavior based on the email format.
- Verified that the form submission works as expected.

## Checklist
- [ ] Code compiles correctly
- [ ] Tests passed successfully
- [ ] Verified that the changes work as expected

Please review the changes and let me know if any further modifications or tests are required. Thank you!

### CODE REVIEW

**Comment by Sarfraz-droid** [comment]
Hi, @Deeppjp116 , Some tests are failing. can you update the empty email test or update it accordingly???

File: .env.sample
Comment (Sarfraz-droid) [inline .env.sample:1]: Don't delete the `.env.sample`. remove this change from your PR

**Review by Sarfraz-droid — CHANGES_REQUESTED** [review]

File: old_local-firebase-emulators-import-export.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (samyakshah3008) [inline old_local-firebase-emulators-import-export.md:1]: Not sure the reason behind this empty line? can you please look into it?

File: src/components/AuthPage/ForgotPassword/index.jsx [hunk 2]
```diff
@@ -16,151 +16,161 @@ import { Alert } from "@mui/material";
...
+  }
 });
 
 const ForgotPassword = ({
-	rootBackground = "rgba(0,0,0,.01)",
-	confirmationText = "We have sent you an email containing the link to reset your password .Please check your inbox including spams",
-	fontweight = "800",
-	buttonColor = "blue",
+  rootBackground = "rgba(0,0,0,.01)",
+  confirmationText = "We have sent you an email containing the link to reset your password .Please check your inbox including spams",
```
Comment (samyakshah3008) [inline src/components/AuthPage/ForgotPassword/index.jsx:32]: Nitpik: 
Can we modify your text as below:
"We have sent you an email containing the link to reset your password. Please check your inbox including spams"

Change: 
Add full stop after password and one gap after full stop

File: src/components/AuthPage/ForgotPassword/index.jsx [hunk 3]
```diff
@@ -16,151 +16,161 @@ import { Alert } from "@mui/material";
...
-	const [open, setOpen] = React.useState(true);
-	const errorProps = useSelector(({ auth }) => auth.profile.error);
-	const loadingProps = useSelector(({ auth }) => auth.profile.loading);
-	const dispatch = useDispatch();
+  const firebase = useFirebase();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+  const [success, setSuccess] = useState(false);
+  const [email, setEmail] = useState("");
+  const [open, setOpen] = React.useState(true);
```
Comment (samyakshah3008) [inline src/components/AuthPage/ForgotPassword/index.jsx:41]: Can we use `useState` as similar to what we did for above other useStates? This is creating code inconsistency I feel, what's your input?

File: src/components/AuthPage/ForgotPassword/index.jsx [hunk 4]
```diff
@@ -16,151 +16,161 @@ import { Alert } from "@mui/material";
...
+        />
+        <Button
+          variant="contained"
+          color="primary"
+          loading={loading}
+          className="mt-10"
+          type="submit"
+          fullWidth
+          data-testId="forgotPasswordButton"
+          disabled={isButtonDisabled} // Disable the button if email format is incorrect
```
Comment (samyakshah3008) [inline src/components/AuthPage/ForgotPassword/index.jsx:146]: Great to see you have added comments, this is improving code quality!

File: src/components/AuthPage/ForgotPassword/index.jsx [hunk 5]
```diff
@@ -16,151 +16,161 @@ import { Alert } from "@mui/material";
...
+        or
+      </Grid>
+      <Grid justify="center" align="center" className="mt-24">
+        <Grid sm={24} className="center">
+          <Link to={"/login"}>Back to Sign in</Link>
+        </Grid>
+      </Grid>
+      <Grid justify="center" align="center" className="mt-24">
+        <Grid sm={24} className="center">
+          New to <span className="brand-font text-bold">CodeLabz</span>?{" "}
```
Comment (samyakshah3008) [inline src/components/AuthPage/ForgotPassword/index.jsx:161]: Can we please remove `{" "}` ?

**Review by samyakshah3008 — APPROVED** [review]
Solid work @Deeppjp116 💪 

I have added a few comments, please check! 
Rest looks good to me! 🚀

**Review by samyakshah3008 — COMMENTED** [review]
Please address to @Sarfraz-droid requested changes.

**Comment by Sarfraz-droid** [comment]
@Deeppjp116 I'm closing this PR for now. How about if you can create a new issue for the same and  discuss in the gitter channel [here](https://matrix.to/#/#scorelab_CodeLabz:gitter.im)?

File: .env.sample
Comment (Deeppjp116) [reply .env.sample:1]: I apologise for the inconvenience, however I changed the file's name to ```.env```. I'll give an update soon.

**Pull request closed without merging**